### PR TITLE
Unify documentation separator

### LINKED ISSUE #1083: Unify documentation to use `-` instead of `_`
### Describe what the documentation is missing.

Some docs use `_` as the separator and some use `-`, for example:
- `https://flower.dev/docs/example_walkthrough_pytorch_mnist.html`
- `https://flower.dev/docs/example-mxnet-walk-through.html`

This is inconsistent.

### Suggest your improvement.

We want to unify all docs to use `-` ("minus") as the only separator. This involves the following TODO's:

- [ ] Rename all docs that use `_` (and replace it with `-`)
- [ ] For each renamed doc, configure a redirect from the old naming (e.g., `quickstart_pytorch`) to the new page (e.g., `quickstart-pytorch`) using, for example, https://documatt.gitlab.io/sphinx-reredirects/index.html
- [ ] Update all references in the docs with the new naming (e.g., in `index.rst`)
- [ ] Update all other references with the new naming (e.g., README.md, code examples, website, ...)

Last, but not least, this should all be done in a single PR. We should not split, for example, the renaming of a page and the configuration of a redirect into two PRs because it would lead to a deployed version of the docs where users would see a 404 page when they use a link that points to the old naming.

### PR DESCRIPTION
**Pull request by RISHIKESHAVAN**
#### Reference Issues/PRs

This PR fixes #1083.

#### What does this implement/fix? Explain your changes.

In order to unify documentation, the docs that contain `_` (underscore) as separator have been renamed to use `-` (minus). Two main changes were implemented for this. 
First, 8 fiilenames have been changed:

1. /doc/source/quickstart_mxnet.rst
2. /doc/source/quickstart_pytorch_lightning.rst
3. /doc/source/example_walkthrough_pytorch_mnist.rst
4. /doc/source/quickstart_huggingface.rst
5. /doc/source/quickstart_pytorch.rst
6. /doc/source/quickstart_tensorflow.rst
7. /doc/source/release_process.rst
8. /doc/source/quickstart_scikitlearn.rst

The references to these files have also been updated, majorly in `README.md` and `index.rst`

Secondly, for each renamed doc redirects were configured in the `conf.py` file using `sphinx_reredirects`. A new extension was also added for the same in the file. This was done to redirect from the old naming (e.g., `quickstart_pytorch`) to the new page (e.g., `quickstart-pytorch`)

#### Any other comments?
-

### CODE REVIEW

**Comment by RISHIKESHAVAN** [comment]
@tanertopal I was able to test it locally using sphinx-build and the changes worked fine. Although I am not sure if testing locally would be enough for this. 

Also, the `Docs/build_docs` action is failing due to the new extension used `sphinx_reredirects`. Any suggestion on what needs to be done for that?

**Review by danieljanes — DISMISSED** [review]
@RISHIKESHAVAN awesome, thanks for the PR, I think it looks great!
@tanertopal I've approved it, what would be the best way to ensure that we don't break anything?

**Comment by tanertopal** [comment]
@danieljanes I am going to test this - this week - and if the redirects work correctly I'll merge it.

**Comment by RISHIKESHAVAN** [comment]
Hi @danieljanes @tanertopal, can you provide some feedback on this PR? Are the changes working fine after the tests?

**Comment by danieljanes** [comment]
Hi @RISHIKESHAVAN , sorry for the delay and thanks for the ping. Busy week(s), I'm on vacation and @tanertopal was occupied with another workstream. Apart from updating your branch with the latest changes from `main` there's nothing to do on your end. I've reviewed the changes and approved them (I'll have to re-approve the updated branch), the only thing missing is testing the deployment (we don't want to break old links). I'm optimistic that we can merge in the next 1-2 weeks.

**Comment by RISHIKESHAVAN** [comment]
> Hi @RISHIKESHAVAN , sorry for the delay and thanks for the ping. Busy week(s), I'm on vacation and @tanertopal was occupied with another workstream. Apart from updating your branch with the latest changes from `main` there's nothing to do on your end. I've reviewed the changes and approved them (I'll have to re-approve the updated branch), the only thing missing is testing the deployment (we don't want to break old links). I'm optimistic that we can merge in the next 1-2 weeks.

@danieljanes 
Thank you. I have updated the branch with the latest changes.

**Review by danieljanes — APPROVED** [review]

File: doc/source/conf.py [hunk 1]
```diff
@@ -61,6 +62,19 @@
...
+# To prevent 404 errors and redirect to the new pages.
+redirects = {
+    "quickstart_mxnet": "quickstart-mxnet",
+    "quickstart_pytorch_lightning": "quickstart-pytorch-lightning",
+    "example_walkthrough_pytorch_mnist": "example-walkthrough-pytorch-mnist",
+    "quickstart_huggingface": "quickstart-huggingface",
+    "quickstart_pytorch": "quickstart-pytorch",
+    "quickstart_tensorflow": "quickstart-tensorflow",
+    "release_process": "release-process",
+    "quickstart_scikitlearn": "quickstart-scikitlearn"
```
Comment (tanertopal) [inline doc/source/conf.py:75]:
```suggestion
    "quickstart_mxnet": "quickstart-mxnet.html",
    "quickstart_pytorch_lightning": "quickstart-pytorch-lightning.html",
    "example_walkthrough_pytorch_mnist": "example-walkthrough-pytorch-mnist.html",
    "quickstart_huggingface": "quickstart-huggingface.html",
    "quickstart_pytorch": "quickstart-pytorch.html",
    "quickstart_tensorflow": "quickstart-tensorflow.html",
    "release_process": "release-process.html",
    "quickstart_scikitlearn": "quickstart-scikitlearn.html"
```

**Review by tanertopal — APPROVED** [review]

**Comment by tanertopal** [comment]
I made a slight fix for the redirects and set it to merge.

**Comment by RISHIKESHAVAN** [comment]
Thank you @tanertopal and @danieljanes !

**Comment by danieljanes** [comment]
Thank you too for making this happen @RISHIKESHAVAN 👍

**Pull request merged**